pkg/host/internal/sriov/sriov.go

```diff
@@ -176,8 +176,16 @@ func (s *sriov) VFIsReady(pciAddr string) (netlink.Link, error) {
 		vfLink, err = s.netlinkLib.LinkByName(vfName)
 		if err != nil {
 			log.Log.Error(err, "VFIsReady(): unable to get VF link", "device", pciAddr)
+			return false, nil
 		}
-		return err == nil, nil
+
+		// try to get the interface name again and check if the name changed
+		vfName = s.networkHelper.TryGetInterfaceName(pciAddr)
```

Comment:
The problem is that `TryGetInterfaceName(pciAddr)` might return `eth0` value that is not unique for the system, and `LinkByName('eth0')` might return a different VF.

I'm not sure this would solve the problem :thinking:

Reply:
so the idea here was
1. getInterface name
2. vfLink object 
3. getInterface name again

and we have two options here:
if the second time it's still eth0 this means the vfLink object we did before contains the right mac address(even if the VF will change name).
if the vf change is name we will try again

Reply:
I think the problem is this:
suppose in the system there are two different VFs named `eth0`  (0000:19:00.5 and 0000:19:00.6) and we are calling `VFIsReady("0000:19:00.5")`

`vfName = s.networkHelper.TryGetInterfaceName("0000:19:00.5")` -> `eth0`
`vfLink, err = s.netlinkLib.LinkByName(vfName)` -> might return `0000:19:00.6`

then 
`vfName = s.networkHelper.TryGetInterfaceName(pciAddr)` still  returns `eth0`

and the function will return the .6 netlink reference, instead of the .5 one. WDYT?

Reply:
oh ok now I get your point the thing is that is not possible on a given time we can only have one interface with the same name that from the kernel lock system. some more info about that from our kernel developer

```
The interface name is assigned by the kernel during netdevice_register(). It is while holding the lock (rtnl_lock) and the kernel performs a name collision check.
It is not possible to have two interfaces with the same name, but when eth0 is renamed to e.g. enp12s23, another device can use eth0 again, but not at the same time (rtnl_lock). The name of the device can be changed later (e.g. with the help of udev and this is also easily done, but again there is rtnl_lock which prevents collisions in names).
```

Reply:
Right!